packages/aws-client/src/S3ObjectStore.ts

```diff
@@ -0,0 +1,79 @@
+import aws from 'aws-sdk';
+import * as querystring from 'querystring';
+import { URL } from 'url';
+import { s3 } from './services';
+import { headObject, parseS3Uri } from './S3';
+
+/**
+ * Class to use when interacting with S3
+ *
+ */
+export class S3ObjectStore {
+  s3: aws.S3;
+
+  /**
+   * Builds and returns an S3 object store
+   */
+  constructor() {
+    this.s3 = s3();
```

Comment:
There's a subtle difference between this and the Harmony code in that the Harmony code seems to support overriding the `s3` class used via the constructor. It might be nice to add that feature for the sake of testing, which would eliminate the need to use sinon

Reply:
Ah yes, I hadn't considered that. Will add.